### PR TITLE
Feature/user registration

### PR DESCRIPTION
**Pull request by swietlikm**
- created Registrtaion View
- created Registration Template
- added tests

### CODE REVIEW

File: conftest.py [hunk 1]
```diff
@@ -4,12 +4,18 @@
...
 from django.contrib.auth import get_user_model
+from django.test import Client
 
 from core.models import Community, Post
 
 User = get_user_model()
 
 
+@pytest.fixture()
+def client() -> type[Client]:
```
Comment (jacoor) [inline conftest.py]: Why? 
we have this: https://pytest-django.readthedocs.io/en/latest/helpers.html#client-django-test-client

File: users/templates/users/registration.html [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+{% load static %}
+
+{% block content %}
+
+    <div class="container">
+        <h3 class="mb-4">Registration</h3>
+        <div class="d-flex justify-content-center">
+            <div class="mx-auto col-10 col-md-8 col-lg-6 vh-100 mt-lg-3">
+                <form method="POST" action=".">
+                    {% csrf_token %}
```
Comment (jacoor) [inline users/templates/users/registration.html:11]: Missing non_field_errors
https://docs.djangoproject.com/en/5.0/ref/forms/api/#django.forms.Form.non_field_errors
https://stackoverflow.com/a/16215790 - it's and old one but still valid. It's a good practice to add non field error handling.

File: users/tests/test_registration.py [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+) -> None:
+    data: dict = {
+        "email": "testuser@example.com",
+        "password1": "Pass2712!",
+        "password2": "Pass2712!",
+    }
+    response = client.post(register_url, data)
+    assert response.status_code == 302
+    assert response.url == reverse("home-page")
+    assert user_model.objects.filter(email="testuser@example.com").exists()
```
Comment (jacoor) [inline users/tests/test_registration.py:37]: Can you prove that this user did not exist earlier?

File: users/tests/test_registration.py [hunk 4]
```diff
@@ -0,0 +1,63 @@
...
+    data: dict = {
+        "email": "",
+        "password1": "",
+        "password2": "",
+    }
+    response = client.post(register_url, data)
+    assert response.status_code == 200
+    assert "users/registration.html" in [t.name for t in response.templates]
+    assert not user_model.objects.filter(email="").exists()
+    assert "This field is required." in response.content.decode()
```
Comment (jacoor) [inline users/tests/test_registration.py]: Which field? Or how many error messages are there? Either way is good. 
Right now this test only show that one of the fields is required, doesn't show which one.

File: users/tests/test_registration.py [hunk 5]
```diff
@@ -0,0 +1,63 @@
...
+    data: dict = {
+        "email": "testuser@example.com",
+        "password1": "TestPassword123",
+        "password2": "DifferentPassword123",
+    }
+    response = client.post(register_url, data)
+    assert response.status_code == 200
+    assert "users/registration.html" in [t.name for t in response.templates]
+    assert not user_model.objects.filter(email="testuser@example.com").exists()
+    assert "The two password fields didn’t match." in response.content.decode()
```
Comment (jacoor) [inline users/tests/test_registration.py]: hmm I would give up on testing passwords, as this is done in UserCreationForm.
But if you really want robust testing, also add check for invalid emain. 

And missing tests for already existing user.

**Review by jacoor — CHANGES_REQUESTED** [review]

File: conftest.py (hunk 1 above)
Comment (swietlikm) [reply conftest.py]: I do not understand, what you linked is reffering to An instance of a django.test.Client and the fixture is exactly this

Comment (jacoor) [reply conftest.py]: I mean, pytest-django, the package we have already, is providing this. We don't need second one in this fixture

**Review by jacoor — APPROVED** [review]

**Pull request merged**